src/vm/contexts.rs

```diff
@@ -966,16 +994,23 @@ impl<'a, 'b> Environment<'a, 'b> {
         &mut self,
         contract_identifier: QualifiedContractIdentifier,
         contract_content: &str,
+        sponsor: Option<PrincipalData>,
```

Comment:
This function should just use `self.sponsor` here rather than taking it in as an argument -- the `Environment` struct should be constructed with the sponsor set to the correct value.

Reply:
(And the same applies to `initialize_contract_from_ast`)